sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -73,17 +74,78 @@ class CollationStringExpressionsSuite extends QueryTest
     })
   }
 
+  test("INSTR check result on explicitly collated strings") {
```

Comment:
try covering some more edge cases, such as: empty strings, uppercase and lowercase mix, different byte-length chars, etc.